src/scr_index.c

```diff
@@ -1674,7 +1672,7 @@ int scr_summary_build(const spath* prefix, const spath* dir, int id)
 
   if (scr_summary_read(dir, summary) != SCR_SUCCESS) {
     /* now only return success if we successfully write the file */
-    int rc = SCR_FAILURE;
+    rc = SCR_FAILURE;
```

Comment:
There seems to be strange code in line 1729 further down which reads:

```
      for (list_elem = kvtree_elem_first(list_hash);
           list_elem != NULL;
           list_elem = kvtree_elem_next(list_elem))
      {
        /* get the hash for this checkpoint */
        kvtree* dset_hash = kvtree_elem_hash(list_elem);
        /* unset the BUILD, MISSING, UNRECOVERABLE, INVALID, and XOR keys for this checkpoint */
for this checkpoint */
        kvtree_unset(dset_hash, SCR_SCAN_KEY_MAP);
        kvtree_unset(dset_hash, SCR_SCAN_KEY_BUILD);
        kvtree_unset(dset_hash, SCR_SCAN_KEY_MISSING);
        kvtree_unset(dset_hash, SCR_SCAN_KEY_UNRECOVERABLE);
        kvtree_unset(dset_hash, SCR_SCAN_KEY_INVALID);
        kvtree_unset(dset_hash, SCR_SCAN_KEY_PARTNER);
        kvtree_unset(dset_hash, SCR_SCAN_KEY_MAPPARTNER);
        kvtree_unset(dset_hash, SCR_SCAN_KEY_XOR);
        kvtree_unset(dset_hash, SCR_SCAN_KEY_MAPXOR);
        /* record the summary file version number */
        kvtree_set_kv_int(dset_hash, SCR_SUMMARY_KEY_VERSION, SCR_SUMMARY_FILE_VERSION_6);
        /* write the summary file out */
        rc = scr_summary_write(prefix, dir, dset_hash);
      }
```

resetting `rc` in each loop iteration, causing the return value to be the `rc` of the last loop iteration.

Seems to either need a `if(rc != SCR_SUCCESS) break;` or so, or given that `list_size` is checked to be 1, no loop at all.
```

Reply:
Fixed in latest push